R/utilities.R

```diff
@@ -0,0 +1,540 @@
+        old_fac_spellings <- read_fac_spellings()
+    }
+    if (is.null(hifld_data)) {
+        hifld_data <- read_hifld_data()
+    }
+    
+    # Add source to HIFLD for coalesce 
+    hifld_data <- hifld_data %>% 
+        mutate(population_source = ifelse(is.na(POPULATION), NA, "HIFLD"), 
+               capacity_source = ifelse(is.na(CAPACITY), NA, "HIFLD"))
+    
+    new_fac_info %>%
+        
+        # Populate from HIFLD 
+        rowwise() %>% 
+        mutate(
+            Population.Feb20 = coalesce(
+                Population.Feb20, 
+                pull_hifld_field(HIFLD.ID, "POPULATION", hifld_data)),
+            Source.Population.Feb20 = coalesce(
+                Source.Population.Feb20, 
+                pull_hifld_field(HIFLD.ID, "population_source", hifld_data)), 
+            Address = coalesce(
+                Address, 
+                pull_hifld_field(HIFLD.ID, "ADDRESS", hifld_data)), 
+            City = coalesce(
+                stringr::str_to_upper(City), 
+                pull_hifld_field(HIFLD.ID, "CITY", hifld_data)), 
+            Zipcode = coalesce(
+                Zipcode, 
+                pull_hifld_field(HIFLD.ID, "ZIP", hifld_data)), 
+            County = coalesce(
+                stringr::str_to_upper(County), 
+                pull_hifld_field(HIFLD.ID, "COUNTY", hifld_data)), 
+            County.FIPS = coalesce(
+                County.FIPS, 
+                pull_hifld_field(HIFLD.ID, "COUNTYFIPS", hifld_data)), 
+            Capacity = coalesce(
+                Capacity, 
+                pull_hifld_field(HIFLD.ID, "CAPACITY", hifld_data)), 
+            Source.Capacity = coalesce(
+                Source.Capacity, 
+                pull_hifld_field(HIFLD.ID, "capacity_source", hifld_data)), 
+            Website = coalesce(
+                Website, 
+                pull_hifld_field(HIFLD.ID, "WEBSITE", hifld_data)), 
+            Latitude = coalesce(
+                Latitude, 
+                pull_hifld_field(HIFLD.ID, "lat", hifld_data)), 
+            Longitude = coalesce(
+                Longitude, 
+                pull_hifld_field(HIFLD.ID, "lon", hifld_data))
+        ) %>% 
+        ungroup() %>%
+        
+        # Geocode addresses
+        tidygeocoder::geocode(
+            method = "geocodio", 
+            street = Address,
+            city = City,
+            county = County,
+            state = State,
+            postalcode = Zipcode,
+            lat = "lat_geo_",
+            long = "long_geo_"
+        ) %>%
+        mutate(
+            Latitude = coalesce(Latitude, lat_geo_),
+            Longitude = coalesce(Longitude, long_geo_)
+        ) %>%
+        select(-lat_geo_, -long_geo_) %>%
+        
+        # Assign Facility.ID
+        mutate(Facility.ID = generate_new_fac_id(
+            old_fac_info = old_fac_info, 
+            old_fac_spellings = old_fac_spellings) + row_number() - 1)
+}
+
+populate_new_spellings <- function(
```

Comment:
I'm curious about the case here when we have added a new facility to fac_info AND fac_spellings on the same day. Will we run into trouble if this function uses `old_fac_info`?

Reply:
That's a super good question! The way I have the `main.R` script set up, you update the info sheet first, and then update the spellings sheet based on the newly updated info sheet (i.e. pass in `old_fac_info = updated_fac_info`). I'll add comments making that more clear, but super open to suggestions for making it less confusing!

Reply:
Sounds good!